### ISSUE TITLE
Option "spellright.configurationScope": "user" is not working properly on Windows 

### ISSUE THREAD

**Issue by Nutznieser** [bartosz-antosik/vscode-spellright#205]
Since some weeks I noticed that the option `"spellright.configurationScope": "user"` is not working properly anymore on Windows. 

My environment is Window 10 and Vscode 1.27.1 and Spellright 2.5.32.

e.g If I have a workspace active and want to change spell checking On or Off for a specific document type, then a new folder `.vscode` with the changed configuration is created in my workspace. BUT the `configurationScope `is set to `user`!

On my Linux system all works right.

Can it be that this problem appears because of the change of the installation root of vscode on Windows systems? Now it is on `%LOCALAPPDATA%\Programs\Microsoft VS Code`.

**Comment by bartosz-antosik** [comment]
It may be! I have not explored the so called "user level" installation but I am aware that there is this new way of installing. Will have to have a look.

Could you try to tell me where is user configuration located in this situation?

**Comment by Nutznieser** [comment]
Thanks for you fast response! ... The user configuration is on the same directory for both installation modes. It is `%APPDATA%\Roaming\Code\User`

**Comment by bartosz-antosik** [comment]
I had a look at the code, and, funny thing (because it was different some time ago) it uses all standard methods provided by VSCode! So I am surprised. Will have to examine deeper. Do not know however how long can it take.

**Comment by Nutznieser** [comment]
I did a more deeper check on my machines. It seems the problem depends not to the installation mode and is not limited only to windows.

#### I found out it will raise under the following circumstance:
If you had already open a workspace **and** open an additional file from outside the workspace scope **and** you switch spell checking On or Off for THIS document, **then** a new folder `.vscode` with the changed configuration is created inside the workspace. ... And that despite of the `configurationScope` is set to `user`! It happens on Windows and Linux.

Is this behavior intended? Maybe you can reproduce and check it.

#### Another point I found out in my investigation:
If I want set spell checking On for a new document type, e.g `log` (for logfiles). If it is not already registered inside `"spellright.parserByClass"`, then spell checking is not possible.

Here a situation to reproduce it. In the settings you have:
```
"spellright.configurationScope": "user",
"spellright.documentTypes": [
    "restructuredtext",
    "text",
    "properties",
    "markdown"
],
"spellright.parserByClass": {
    "properties": {
        "parser": "plain"
    }
}
```
So, you open a logfile (document type=log) and intends to set spell checking On for this document. Then spellright ask you first for the language and second for the document type. But after you set your choices, nothing happens. ... Maybe you can check this problem too. 

Thanks for your efforts!

**Comment by bartosz-antosik** [comment]
@Nutznieser Do I suspect correctly that `spellright.configurationScope` is set in User settings?

**Comment by Nutznieser** [comment]
@bartosz-antosik Yes, that is right.

**Comment by bartosz-antosik** [comment]
Both the problems described two posts above were real and they should be resolved in just released version 3.0.0.

**Comment by Nutznieser** [comment]
Thanks very much for solving these issues! Works perfect!
